dask/dataframe/core.py

```diff
@@ -2209,24 +2209,31 @@ def info(self, buf=None, verbose=False, memory_usage=False):
         Concise summary of a Dask DataFrame.
         """
 
-        from pandas.formats.format import _put_lines
-
-        lines = [str(type(self))]
-
         if buf is None:
             import sys
             buf = sys.stdout
 
+        lines = [str(type(self))]
+
         if len(self.columns) == 0:
             lines.append('Index: 0 entries')
             lines.append('Empty %s' % type(self).__name__)
-            _put_lines(buf, lines)
+            pd.formats.format._put_lines(buf, lines)
             return
 
+        # Group and execute the required computations
+        computations = {}
+        if verbose:
+            computations.update({'index': self.index, 'count': self.count()})
+        if memory_usage:
+            computations.update({'memory_usage': self.map_partitions(pd.DataFrame.memory_usage, index=True)})
+        computations = dict(zip(computations.keys(), da.compute(*computations.values())))
```

Comment:
This doesn't seem overly ugly to me.  Are we certain that keys and values always iterate in the same order?